invokeai/frontend/web/src/features/gallery/store/types.ts

```diff
@@ -22,6 +22,7 @@ export type GalleryState = {
   limit: number;
   starredFirst: boolean;
   orderDir: OrderDir;
+  searchTerm?: string;
```

Comment:
This can be `searchTerm: string;` , then add `searchTerm: ''` to the initial state. The persistence layer will handle adding the missing `searchTerm` state to the slice when hydrating it.

Then, in the component, we don't need to check if it's undefined.